### PR TITLE
Fix #20: Revamp User Controller 

### LINKED ISSUE #20: Revamp User Controller
## Issue Description
Refactor the User Controller by removing the use of authentication.getAuthenticated() and implement authentication handling using try-catch blocks.

### Expected Behavior
The User Controller should robustly manage authentication errors through proper exception handling, enhancing overall stability.

### Actual Behavior
The current implementation relies on authentication.getAuthenticated(), which may not adequately handle exceptions.

### PR DESCRIPTION
**Pull request by JishnuGoyal**
# Quash Pull Request

## Description
This pull request refactors the `UserController` by replacing the use of authentication.isAuthenticated() with exception handling using try-catch blocks. The goal is to enhance the robustness and stability of the controller by ensuring proper handling of potential authentication-related errors.

Fixes #20 

## Type of Change
Please check the relevant option(s)

- [x] 🐛 Bug fix (non-breaking change addressing an issue)
- [ ] ✨ New feature (non-breaking change adding functionality)
- [ ] 💥 Breaking change (fix or feature causing existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [x] 🧹 Code refactor
- [ ] 🔧 Build configuration change
- [ ] ⚡ Performance improvement
- [ ] 🧪 Test update

## Component
Please check the relevant option(s)

- [ ] 📱 Android SDK
- [x] 🖥️ Backend
- [ ] 🌐 Frontend
- [ ] 🔄 Cross-component change

## Changes Made
Provide a bullet-point list of the key changes made in this PR

- Replaced the existing logic that relied on authentication.isAuthenticated() with try-catch blocks.
- Added `NullPointerException` to handle cases where the authentication object might be null.
- A general Exception catch block is added to handle any other unexpected errors.
- When exceptions are caught, appropriate error messages are returned to the client, with specific HTTP status codes (e.g., UNAUTHORIZED, INTERNAL_SERVER_ERROR).
- 

## How Has This Been Tested?
Describe the tests you ran to verify your changes. Provide instructions to reproduce and list any relevant details for your test configuration.

- [ ] Unit tests
- [ ] Integration tests
- [ ] Manual tests

**Test Configuration**:
- Quash SDK Version:
- Device/Environment:
- OS Version:
- Backend Version (if applicable):
- Frontend Version (if applicable):

## Checklist
Please check all applicable items

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in complex areas
- [ ] I have updated the documentation accordingly
- [ ] My changes generate no new warnings or errors
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Additional Notes
Please review the changes and provide feedback. This refactor should not affect the functionality of the controller but is intended to make the code more robust and maintainable.

## Screenshots (if appropriate)
If your changes include visual updates, please add before/after screenshots here.

### CODE REVIEW

File: backend/src/main/java/com/quashbugs/quash/controller/UserController.java [hunk 1]
```diff
@@ -74,16 +69,14 @@ public ResponseEntity<ResponseDTO> getUserVerified(Authentication authentication
      */
     @PatchMapping
     public ResponseEntity<ResponseDTO> updateUser(Authentication authentication, @RequestBody User updatedUser) {
-        if (authentication != null && authentication.isAuthenticated()) {
+        try {
             User user = (User) authentication.getPrincipal();
             userService.updateUser(user, updatedUser);
```
Comment (JishnuGoyal) [inline backend/src/main/java/com/quashbugs/quash/controller/UserController.java:74]: What happens in this case when the "user" object is null? when the "user" object is null, will the client know that no action has been taken?

**Review by JishnuGoyal — COMMENTED** [review]
Taking a self review pass.

**Comment by JishnuGoyal** [comment]
Hi @ironmark3 @Hemanthr1 , for this issue, my approach involves handling the exception where "authentication" object is null and where the "user" object retrieved is null. Additionally, I have added an exception handling for any other unexpected exception. Please provide feedback on the same.

This is my first pull request on this project - apologies in advance for something that i miss, though I tried to cover most things. Let me know where things can be improved :)

**Comment by JishnuGoyal** [comment]
@ironmark3 @Hemanthr1 I would like to let you know that I haven't been able to test these changes on my local machine, because when I tried to run `mvn clean install`, or `mvn spring-boot:run`, I got 

> `release version 17 not supported`

 error:  
 
image

image

image

**Do you have any insight why this might occur?**

**Comment by Hemanthr1** [comment]
@siffat-quash Review this PR and help with testing locally

**Comment by JishnuGoyal** [comment]
jishnu@jishnus-ubuntu:~/opensource/quash-max$ cd backend
jishnu@jishnus-ubuntu:~/opensource/quash-max/backend$ mvn clean install
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for com.quashbugs:quash:jar:0.0.1-SNAPSHOT
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: org.springframework.boot:spring-boot-starter-security:jar -> duplicate declaration of version (?) @ line 82, column 21
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: org.projectlombok:lombok:jar -> version (?) vs edge-SNAPSHOT @ line 142, column 21
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] 
[INFO] ------------------------< com.quashbugs:quash >-------------------------
[INFO] Building quash 0.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:3.2.0:clean (default-clean) @ quash ---
[INFO] Deleting /home/jishnu/opensource/quash-max/backend/target
[INFO] 
[INFO] --- maven-resources-plugin:3.3.1:resources (default-resources) @ quash ---
[INFO] Copying 1 resource from src/main/resources to target/classes
[INFO] Copying 5 resources from src/main/resources to target/classes
[INFO] 
[INFO] --- maven-compiler-plugin:3.11.0:compile (default-compile) @ quash ---
[INFO] Changes detected - recompiling the module! :source
[INFO] Compiling 115 source files with javac [debug release 17] to target/classes
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.086 s
[INFO] Finished at: 2024-09-03T21:45:08+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:compile (default-compile) on project quash: Fatal error compiling: error: release version 17 not supported -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

**Comment by dhairya-quash** [comment]
@JishnuGoyal I have resolved the build errors and have also tested your changes. We will be merging your changes.

**Pull request merged**